### ISSUE TITLE
Requirements.txt - Dependencies always grab the latest version

### ISSUE THREAD

**Issue by PulsarEtcher** [youngquan/pip-download#27]
If we have a `requirements.txt` with these two lines:

```
django-debug-toolbar==3.2
django==3.1.6
```

django-debug-toolbar has a dependency of "Django>=2.2", which means django==3.1.6 should be enough to satisfy it. However, when pip download starts to download django-debug-toolbar, rather than install django==3.1.6, it grabs the latest version.

**Comment by HUALIMUGU** [comment]
it seems the procedure deal with packages line by line rather than globally analysis.

**Comment by youngquan** [comment]
Yep, this tool is intended to use in offline scenarios. It means that you have to download some packages when the Internet is available and import them to a computer with no Internet connection. So, I think it's better to download more packages than less. However, I do wonder whether this behavior should be changed? Should we only download packages that satisfy the requirements? @PulsarEtcher @dw763j

**Comment by dw763j** [comment]
> Yep, this tool is intended to use in offline scenarios. It means that you have to download some packages when the Internet is available and import them to a computer with no Internet connection. So, I think it's better to download more packages than less. However, I do wonder whether this behavior should be changed? Should we only download packages that satisfy the requirements? @PulsarEtcher @dw763j

In my case, I would agree with download more than less, if someone need to package usable environ, he can just use conda.
